libcontainer/init_linux.go

```diff
@@ -499,6 +499,21 @@ func signalAllProcesses(m cgroups.Manager, s os.Signal) error {
 	if err := m.Freeze(configs.Frozen); err != nil {
 		logrus.Warn(err)
 	}
+	// If signal is SIGKILL try manager.Kill()
+	// in order to leverage "cgroup.kill" instead
+	// of iterating over each process. "cgroup.kill"
+	// was added in Linux kernel 5.14 onwards.
+	if s == unix.SIGKILL {
+		// If manager.Kill() failed
+		// fallback to legacy method
+		err := m.Kill()
+		// If err is nil
+		// return early from function
+		if err == nil {
+			return nil
+		}
+	}
+
```

Comment:
All this is too verbose, and could just be
```suggestion
if s == unix.SIGKILL && m.Kill() == nil {
    return nil
}
// Fallback to slow method.
```